### PR TITLE
Add Function Creation Dialog

### PR DESCRIPTION
**Pull request by Flanker32**
Add Function Creation Dialog

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/forms/CreateArmStorageAccountForm.java [hunk 1]
```diff
@@ -164,16 +164,14 @@ public void itemStateChanged(ItemEvent itemEvent) {
         accoountKindCombo.setRenderer(new ListCellRendererWrapper<Kind>() {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/forms/CreateArmStorageAccountForm.java]: accoountKindCombo typo

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/IntelliJFunctionContext.java [hunk 2]
```diff
@@ -175,6 +175,10 @@ public void setAppServicePlanName(String appServicePlanName) {
         this.appServicePlanName = appServicePlanName;
     }
 
+    public void setAppServicePlanResourceGroup(String appServicePlanResourceGroup) {
+        this.appServicePlanName = appServicePlanResourceGroup;
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/IntelliJFunctionContext.java]: appServicePlanName ??

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/FunctionDeploymentPanel.java [hunk 3]
```diff
@@ -233,6 +236,23 @@ private void onFunctionSelected() {
...
+                final FunctionApp functionApp = dialog.getCreatedWebApp();
+                if (functionApp != null) {
+                    functionDeployConfiguration.setFunctionId(functionApp.id());
+                    presenter.loadFunctionApps(true);
+                } else {
+                    presenter.loadFunctionApps(false);
+                }
+            }
+        });
+        dialog.setVisible(true);
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/FunctionDeploymentPanel.java]: dialog.show?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+    private SubscriptionPanel subscriptionPanel;
+    private AppServicePlanPanel appServicePlanPanel;
+    private AppSettingsTable appSettingsTable;
+
+    private IntelliJFunctionContext functionConfiguration;
+    private FunctionApp result = null;
+    private Project project;
+
+    public FunctionCreationDialog(Project project) {
+        this.project = project;
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java]: Don't need to handle esc?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/IntelliJFunctionContext.java (hunk 2 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/IntelliJFunctionContext.java]: Yes, it should be appServicePlanResourceGroup, thanks~

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/forms/CreateArmStorageAccountForm.java (hunk 1 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/forms/CreateArmStorageAccountForm.java]: FIxed

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java (hunk 4 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java]: Added

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+                        validateConfiguration();
+                    }
+                });
+            } else if (component instanceof Container) {
+                addValidationListener((Container) component, actionListener);
+            }
+        }
+    }
+
+    private void selectOS(boolean isWindows) {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java]: param isWindows is not used?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+        setContentPane(contentPanel);
+        getRootPane().setDefaultButton(buttonOK);
+
+        this.functionConfiguration = new IntelliJFunctionContext(project);
+
+        buttonOK.addActionListener(e -> onOK());
+        buttonCancel.addActionListener(e -> onCancel());
+
+        rdoLinuxOS.addActionListener(e -> selectOS(false));
+        rdoWindowsOS.addActionListener(e -> selectOS(false));
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java]: selectOS(true)?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java (hunk 5 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java]: This parameter has been out of date and should be removed

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java (hunk 6 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/deploy/ui/creation/FunctionCreationDialog.java]: Fixed

**Pull request merged**